frontend/src/authenticated/UserProfile.js

```diff
@@ -0,0 +1,28 @@
+import React, { useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import ProfilePosts from "./ProfilePosts";
+import { saveCurrentUserPosts } from "../store/actions/actions";
+
+const UserProfile = props => {
+  const user = useSelector(state => state.currentUser);
+  const userPosts = useSelector(state => state.currentUserPosts);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (user) {
+      dispatch(saveCurrentUserPosts(user.uuid));
+    }
+  }, [dispatch, user]);
+
+  return (
+    <>
+      {userPosts ? (
+        userPosts.map(post => <ProfilePosts post={post} />)
```

Comment:
Are you sure you don't need key after mapping ???

Reply:
done